xio-core/src/main/java/com/xjeffrose/xio/config/ConfigReloader.java

```diff
@@ -72,24 +87,60 @@ private Config parse(Reader reader) {
     return load(new File(file));
   }
 
-  private boolean performUpdate(Meta<T> oldValue, Meta<T> newValue) {
-    return !MessageDigest.isEqual(oldValue.digest, newValue.digest);
+  private boolean performUpdate(ConfigFileMetadata current, ConfigFileMetadata previous) {
+    return (current.lastModified > previous.lastModified
+        && !MessageDigest.isEqual(current.digest, previous.digest));
+  }
+
+  private ConfigFileMetadata loadMetaData(File configFile) throws IllegalStateException {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      DigestInputStream digester = new DigestInputStream(new FileInputStream(configFile), digest);
+      ConfigFactory.parseReader(new InputStreamReader(digester));
+      return new ConfigFileMetadata(
+          configFile.getAbsolutePath(), configFile.lastModified(), digest.digest());
+    } catch (FileNotFoundException e) {
+      throw new IllegalStateException(
+          "Couldn't load config from file '" + configFile.getAbsolutePath() + "'", e);
+    } catch (NoSuchAlgorithmException e) {
+      throw new IllegalStateException(
+          "Illegal digest algorithm used on file '" + configFile.getAbsolutePath() + "'", e);
+    }
+  }
+
+  private boolean haveWatchFilesChanged() {
+    Boolean filesHaveChanged = false;
+    try {
+      Set<String> keys = new HashSet<String>(watchFiles.keySet());
```

Comment:
"nice to have": java streams api would be "nicer" here

Reply:
eg
```java
List<ConfigFileMetadata> toUpdateList = watchFiles.keySet().stream()
      .map(filePath -> loadMetaData(new File(filePath)))
      .filter(md -> {
        ConfigFileMetadata previousConfigMetadata = watchFiles.get(md.path);
        return performUpdate(md, previousConfigMetadata);
      })
      .collect(Collectors.toList());

    toUpdateList.forEach(update -> watchFiles.put(update.path, update));

    return !toUpdateList.isEmpty();
```

Reply:
k